scripts/in_container/run_generate_constraints.py

```diff
@@ -0,0 +1,460 @@
+    r = requests.get(constraints_url, timeout=60)
+    r.raise_for_status()
+    with config_params.latest_constraints_file.open("w") as constraints_file:
+        constraints_file.write(r.text)
+    console.print(f"[green]Downloaded constraints file from {constraints_url} to {constraints_file.name}")
+
+
+def diff_constraints(config_params: ConfigParams) -> None:
+    """
+    Diffs constraints files and prints the diff to the console.
+    """
+    console.print("[bright_blue]Diffing constraints files")
+    result = run_command(
+        [
+            "diff",
+            "--ignore-matching-lines=#",
+            "--color=always",
+            config_params.latest_constraints_file.as_posix(),
+            config_params.current_constraints_file.as_posix(),
+        ],
+        # always shows output directly in CI without folded group
+        github_actions=False,
+        check=False,
+    )
+    if result.returncode == 0:
+        console.print("[green]No changes in constraints files. exiting")
+        config_params.current_constraints_file.unlink(missing_ok=True)
+        sys.exit(0)
+    result = run_command(
+        [
+            "diff",
+            "--ignore-matching-lines=#",
+            "--color=never",
+            config_params.latest_constraints_file.as_posix(),
+            config_params.current_constraints_file.as_posix(),
+        ],
+        github_actions=config_params.github_actions,
+        check=False,
+        text=True,
+        capture_output=True,
+    )
+    with config_params.constraints_diff_file.open("w") as diff_file:
+        diff_file.write(
+            f"Dependencies {config_params.airflow_constraints_mode} updated "
+            f"for Python {config_params.python}\n\n"
+        )
+        diff_file.write("```diff\n")
+        diff_file.write(result.stdout)
+        diff_file.write("```\n")
+    console.print(f"[green]Diff generated to file: {config_params.constraints_diff_file}")
+
+
+def uninstall_all_packages(config_params: ConfigParams):
+    console.print("[bright_blue]Uninstall All PIP packages")
+    result = run_command(
+        ["pip", "freeze"],
+        github_actions=config_params.github_actions,
+        cwd=AIRFLOW_SOURCES,
+        text=True,
+        check=True,
+        capture_output=True,
+    )
+    all_installed_packages = [
+        dep.split("==")[0]
+        for dep in result.stdout.strip().split("\n")
+        if not dep.startswith(("apache-airflow", "apache-airflow==", "/opt/airflow", "#", "-e"))
+    ]
+    run_command(
+        ["pip", "uninstall", "--root-user-action", "ignore", "-y", *all_installed_packages],
+        github_actions=config_params.github_actions,
+        cwd=AIRFLOW_SOURCES,
+        text=True,
+        check=True,
+    )
+
+
+def get_core_airflow_dependencies(config_params) -> list[str]:
+    result = run_command(
+        ["python", "-c", "import setup; print(','.join(setup.CORE_EXTRAS_DEPENDENCIES.keys()))"],
```

Comment:
If we're now using Python script here, should we just import setup and run it directly instead of running through subprocess?

Reply:
Actually yes :). Good catch.

Reply:
updated both.